client/src/components/selects/addResourceSelect.tsx

```diff
@@ -1,46 +1,32 @@
 import * as React from 'react';
-import { Intent, MenuItem, Position, Menu } from '@blueprintjs/core';
-import {
-  ItemPredicate,
-  ItemRenderer,
-  ItemListRenderer
-} from '@blueprintjs/select';
+import {ResourceDefinition} from'@arkhn/fhir.ts'
+import { Intent, Position } from '@blueprintjs/core';
+import { ItemPredicate, ItemRenderer } from '@blueprintjs/select';
 import { IconName } from '@blueprintjs/icons';
 
 import UploadProfile from 'components/uploads/uploadProfile';
 import TSelect from './TSelect';
+import AddProfileSelect from './addProfileSelect';
 
 interface Props {
   disabled?: boolean;
   icon?: IconName;
-  inputItem?: Resource;
+  inputItem?: ResourceDefinition;
   intent?: Intent;
-  items?: Resource[];
+  items?: ResourceDefinition[];
   loading?: boolean;
   onChange: any;
   popoverProps?: any;
+  fetchProfiles: (r: ResourceDefinition) => Promise<ResourceDefinition[]>;
 }
 
-interface Profile {
-  id: string;
-  name: string;
-  type: string;
-  publisher?: string;
-}
-interface Resource {
-  id: string;
-  name: string;
-  type: string;
-  profiles: Profile[];
-}
-
-const filterByName: ItemPredicate<Profile> = (query, resource: Profile) => {
-  return resource.name.toLowerCase().indexOf(query.toLowerCase()) >= 0;
+const filterByName: ItemPredicate<ResourceDefinition> = (query, definition: ResourceDefinition) => {
+  return definition.name.toLowerCase().indexOf(query.toLowerCase()) >= 0;
 };
 
 const sortAlphabetically = (
```

Comment:
We could refactor this using `localeCompare`, I think Teva did that in some other components

Reply:
done